### PR TITLE
Feature/33595 document definition roles

### PR DESCRIPTION
**Pull request by theo-ritense**

### CODE REVIEW

File: document/src/main/resources/config/liquibase/changelog/20220215-create-document-definition-role-changelog.xml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
```
Comment (Klaas-Ritense) [inline document/src/main/resources/config/liquibase/changelog/20220215-create-document-definition-role-changelog.xml:1]: Existing project that migrate to this version, will they keep seeing their cases?

Do we need to add a migration script that will make sure that all users of existing projects will keep seeing all cases?

**Review by Klaas-Ritense — CHANGES_REQUESTED** [review]

File: document/src/main/resources/config/liquibase/changelog/20220215-create-document-definition-role-changelog.xml (hunk 1 above)
Comment (Rick-Ritense) [reply document/src/main/resources/config/liquibase/changelog/20220215-create-document-definition-role-changelog.xml:1]: They will always see them as an administrator.
But no by default all the cases will be disabled for all roles. So the admin must set the correct roles before they will be visible for a ROLE_USER

File: document/src/main/java/com/ritense/document/domain/DocumentDefinitionRole.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2015-2020 Ritense BV, the Netherlands.
```
[the remark this answers is not in the document]
Comment (theo-ritense) [reply document/src/main/java/com/ritense/document/domain/DocumentDefinitionRole.java:2]: I guess we should update the whole project then. Not part of this story.

File: document/src/main/java/com/ritense/document/autoconfigure/DocumentAutoConfiguration.java [hunk 3]
```diff
@@ -79,11 +81,13 @@ public JsonSchemaDocumentService documentService(
...
         final ResourceLoader resourceLoader,
-        final DocumentDefinitionRepository<JsonSchemaDocumentDefinition> documentDefinitionRepository
+        final DocumentDefinitionRepository<JsonSchemaDocumentDefinition> documentDefinitionRepository,
+        final DocumentDefinitionRoleRepository<JsonSchemaDocumentDefinitionRole> documentDefinitionRoleRepository
     ) {
         return new JsonSchemaDocumentDefinitionService(
             resourceLoader,
-            documentDefinitionRepository
+            documentDefinitionRepository,
+            documentDefinitionRoleRepository
```
[the remark this answers is not in the document]
Comment (theo-ritense) [reply document/src/main/java/com/ritense/document/autoconfigure/DocumentAutoConfiguration.java:90]: Had a discussion with @ThomasRitense about this. I believe there's no way to implement this cleanly without having a breaking change. Plus: the repositories are autowired and should not be breaking on a standard configuration.

File: document/src/main/resources/config/liquibase/changelog/20220215-create-document-definition-role-changelog.xml (hunk 1 above)
Comment (theo-ritense) [reply document/src/main/resources/config/liquibase/changelog/20220215-create-document-definition-role-changelog.xml:1]: I added a migration script for existing installations, where ROLE_USER is assigned to the document definitions.
Any new definitions added during or after the upgrade will not have any role assigned by default.

**Review by Klaas-Ritense — APPROVED** [review]

**Pull request merged**